dynawo/sources/Solvers/SolverSIM/DYNSolverSIM.cpp

```diff
@@ -214,17 +213,96 @@ SolverSIM::calculateIC() {
   solverKIN_->clean();
 }
 
-void SolverSIM::solve(double /*tAim*/, double& tNxt, bool &algebraicModeFound, bool& discreteVariableChangeFound) {
-  bool redoStep = true;
-  int counter = 0;
+void SolverSIM::solve(double /*tAim*/, double& tNxt) {
+  if (recalculateStep_)
+    solveWithStepRecalculation(tNxt);
+  else
+    solveWithoutStepRecalculation(tNxt);
+}
 
-  // Save the initial values before the step
+void SolverSIM::solveWithoutStepRecalculation(double& tNxt) {
+  int counter = 0;
```

Comment:
Copy paste is bad: is there a way to not duplicate the code here?

Reply:
Done